pandas/tests/groupby/test_groupby.py

```diff
@@ -2057,3 +2057,37 @@ def test_groups_repr_truncates(max_seq_items, expected):
 
         result = df.groupby(np.array(df.a)).groups.__repr__()
         assert result == expected
+
+
+def test_groupby_column_index_name_lost():
+    # GH: 29764 groupby loses index sometimes
+    df = pd.DataFrame([[1]], columns=pd.Index(["a"], name="idx"))
```

Comment:
can you parameterize over the functions, you don't need to assert the results, rather assert the column name

Reply:
I've parametrized the tests and merged master